pkg/crypto/ciphersuite/ciphersuite_test.go

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2023 The Pion community <https://pion.ly>
+// SPDX-License-Identifier: MIT
+
+// Package ciphersuite provides the crypto operations needed for a DTLS CipherSuite
+package ciphersuite
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pion/dtls/v2/pkg/protocol"
+	"github.com/pion/dtls/v2/pkg/protocol/recordlayer"
+)
+
+func TestGenerateAEADAdditionalDataCID(t *testing.T) {
+	cases := map[string]struct {
+		reason     string
+		header     *recordlayer.Header
+		payloadLen int
+		expected   []byte
+	}{
+		"WithConnectionID": {
+			reason: "Should successfully generate additional data with valid header",
+			header: &recordlayer.Header{
+				ContentType:    protocol.ContentTypeConnectionID,
+				ConnectionID:   []byte{1, 2, 3, 4, 5, 6, 7, 8},
+				Version:        protocol.Version1_2,
+				Epoch:          2,
+				SequenceNumber: 277,
+			},
+			payloadLen: 1784,
+			expected:   []byte{255, 255, 255, 255, 255, 255, 255, 255, 25, 8, 25, 254, 253, 0, 2, 0, 0, 0, 0, 1, 21, 1, 2, 3, 4, 5, 6, 7, 8, 6, 248},
```

Comment:
would it be reasonable to prove that these are the same by not just checking the bytes but also unmarshaling them and checking the structure? I only say this because it could help prove that the bytes aren't just copied over from test output that could be wrong while an unmarshal proves with little doubt that the bytes are good.

Reply:
Gotcha. Good point